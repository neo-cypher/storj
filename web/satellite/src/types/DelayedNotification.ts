// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { VNode, createTextVNode } from 'vue';

import { getId } from '@/utils/idGenerator';

import SuccessIcon from '@/../static/images/notifications/success.svg';
import NotificationIcon from '@/../static/images/notifications/notification.svg';
import ErrorIcon from '@/../static/images/notifications/error.svg';
import WarningIcon from '@/../static/images/notifications/warning.svg';

export enum NotificationType {
    Success = 'Success',
    Info = 'Info',
    Error = 'Error',
    Warning = 'Warning',
}

type RenderFunction = () => (string | VNode | (string | VNode)[]);
export type NotificationMessage = string | RenderFunction;

const StyleInfo: Record<NotificationType, { icon: string; backgroundColor: string }> = {
    [NotificationType.Success]: {
        backgroundColor: '#DBF1D3',
        icon: SuccessIcon,
    },
    [NotificationType.Error]: {
        backgroundColor: '#FFD4D2',
        icon: ErrorIcon,
    },
    [NotificationType.Warning]: {
        backgroundColor: '#FCF8E3',
        icon: WarningIcon,
    },
    [NotificationType.Info]: {
        backgroundColor: '#D0E3FE',
        icon: NotificationIcon,
    },
};

export class DelayedNotification {
    public readonly id: string;

    private readonly callback: () => void;
    private timerId: ReturnType<typeof setTimeout>;
    private startTime: number;
    private remainingTime: number;

    public readonly type: NotificationType;
    public readonly title: string | undefined;
    public readonly messageNode: RenderFunction;
    public readonly backgroundColor: string;
    public readonly icon: string;

    constructor(callback: () => void, type: NotificationType, message: NotificationMessage, title?: string) {
        this.callback = callback;
        this.type = type;
        this.title = title;
        this.messageNode = typeof message === 'string' ? () => createTextVNode(message) : message;
        this.id = getId();
        this.remainingTime = 3000;
        this.start();

        this.backgroundColor = StyleInfo[type].backgroundColor;
        this.icon = StyleInfo[type].icon;
    }

    public pause(): void {
        clearTimeout(this.timerId);
        this.remainingTime -= new Date().getMilliseconds() - this.startTime;
    }

    public start(): void {
        this.startTime = new Date().getMilliseconds();
        this.timerId = setTimeout(this.callback, this.remainingTime);
    }
}
