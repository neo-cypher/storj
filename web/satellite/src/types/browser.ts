// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

import { Component } from 'vue';

import { BrowserObject } from '@/store/modules/objectBrowserStore';

import RedditIcon from '@poc/components/icons/share/IconReddit.vue';
import FacebookIcon from '@poc/components/icons/share/IconFacebook.vue';
import TwitterIcon from '@poc/components/icons/share/IconTwitter.vue';
import HackerNewsIcon from '@poc/components/icons/share/IconHackerNews.vue';
import LinkedInIcon from '@poc/components/icons/share/IconLinkedIn.vue';
import TelegramIcon from '@poc/components/icons/share/IconTelegram.vue';
import WhatsAppIcon from '@poc/components/icons/share/IconWhatsApp.vue';
import EmailIcon from '@poc/components/icons/share/IconEmail.vue';

import imageIcon from '@poc/assets/icon-image-tonal.svg';
import videoIcon from '@poc/assets/icon-video-tonal.svg';
import audioIcon from '@poc/assets/icon-audio-tonal.svg';
import textIcon from '@poc/assets/icon-text-tonal.svg';
import pdfIcon from '@poc/assets/icon-pdf-tonal.svg';
import zipIcon from '@poc/assets/icon-zip-tonal.svg';
import spreadsheetIcon from '@poc/assets/icon-spreadsheet-tonal.svg';
import folderIcon from '@poc/assets/icon-folder-tonal.svg';
import fileIcon from '@poc/assets/icon-file-tonal.svg';

export enum ShareOptions {
    Reddit = 'Reddit',
    Facebook = 'Facebook',
    Twitter = 'Twitter',
    HackerNews = 'Hacker News',
    LinkedIn = 'LinkedIn',
    Telegram = 'Telegram',
    WhatsApp = 'WhatsApp',
    Email = 'E-Mail',
}

export interface ShareButtonConfig {
    color: string;
    getLink: (linksharingURL: string) => string;
    icon: Component;
}

export const SHARE_BUTTON_CONFIGS: Record<ShareOptions, ShareButtonConfig> = {
    [ShareOptions.Reddit]: {
        color: '#5f99cf',
        getLink: url => `https://reddit.com/submit/?url=${url}&resubmit=true&title=Shared%20using%20Storj%20Decentralized%20Cloud%20Storage`,
        icon: RedditIcon,
    },
    [ShareOptions.Facebook]: {
        color: '#3b5998',
        getLink: url => `https://facebook.com/sharer/sharer.php?u=${url}`,
        icon: FacebookIcon,
    },
    [ShareOptions.Twitter]: {
        color: '#55acee',
        getLink: url => `https://twitter.com/intent/tweet/?text=Shared%20using%20Storj%20Decentralized%20Cloud%20Storage&url=${url}`,
        icon: TwitterIcon,
    },
    [ShareOptions.HackerNews]: {
        color: '#f60',
        getLink: url => `https://news.ycombinator.com/submitlink?u=${url}&t=Shared%20using%20Storj%20Decentralized%20Cloud%20Storage`,
        icon: HackerNewsIcon,
    },
    [ShareOptions.LinkedIn]: {
        color: '#0077b5',
        getLink: url => `https://www.linkedin.com/shareArticle?mini=true&url=${url}&title=Shared%20using%20Storj%20Decentralized%20Cloud%20Storage&summary=Shared%20using%20Storj%20Decentralized%20Cloud%20Storage&source=${url}`,
        icon: LinkedInIcon,
    },
    [ShareOptions.Telegram]: {
        color: '#54a9eb',
        getLink: url => `https://telegram.me/share/url?text=Shared%20using%20Storj%20Decentralized%20Cloud%20Storage&url=${url}`,
        icon: TelegramIcon,
    },
    [ShareOptions.WhatsApp]: {
        color: '#25d366',
        getLink: url => `whatsapp://send?text=Shared%20using%20Storj%20Decentralized%20Cloud%20Storage%20${url}`,
        icon: WhatsAppIcon,
    },
    [ShareOptions.Email]: {
        color: '#777',
        getLink: url => `mailto:?subject=Shared%20using%20Storj%20Decentralized%20Cloud%20Storage&body=${url}`,
        icon: EmailIcon,
    },
};

export enum ShareType {
    File = 'File',
    Folder = 'Folder',
    Bucket = 'Bucket',
}

export enum PreviewType {
    None,
    Text,
    CSV,
    Image,
    Video,
    Audio,
    PDF,
}

export const EXTENSION_PREVIEW_TYPES = new Map<string[], PreviewType>([
    [['txt', 'md', 'json', 'xml'], PreviewType.Text],
    [['csv'], PreviewType.CSV],
    [['bmp', 'svg', 'jpg', 'jpeg', 'png', 'ico', 'gif', 'webp'], PreviewType.Image],
    [['m4v', 'mp4', 'webm', 'mov', 'mkv', 'ogv'], PreviewType.Video],
    [['m4a', 'mp3', 'wav', 'ogg', 'aac', 'flac'], PreviewType.Audio],
    [['pdf'], PreviewType.PDF],
]);

export type BrowserObjectTypeInfo = {
    title: string;
    icon: string;
};

/**
 * Contains extra information to aid in the display, filtering, and sorting of browser objects.
 */
export type BrowserObjectWrapper = {
    browserObject: BrowserObject;
    typeInfo: BrowserObjectTypeInfo;
    lowerName: string;
    ext: string;
};

export const EXTENSION_INFOS: Map<string[], BrowserObjectTypeInfo> = new Map([
    [['bmp', 'svg', 'jpg', 'jpeg', 'png', 'ico', 'gif', 'webp'], { title: 'Image', icon: imageIcon }],
    [['m4v', 'mp4', 'webm', 'mov', 'mkv', 'ogv', 'avi'], { title: 'Video', icon: videoIcon }],
    [['m4a', 'mp3', 'wav', 'ogg', 'aac', 'flac', 'aiff'], { title: 'Audio', icon: audioIcon }],
    [['txt', 'docx', 'doc', 'pages'], { title: 'Text', icon: textIcon }],
    [['pdf'], { title: 'PDF', icon: pdfIcon }],
    [['zip'], { title: 'ZIP', icon: zipIcon }],
    [['xls', 'numbers', 'csv', 'xlsx', 'tsv'], { title: 'Spreadsheet', icon: spreadsheetIcon }],
]);
export const FOLDER_INFO: BrowserObjectTypeInfo = { title: 'Folder', icon: folderIcon };
export const FILE_INFO: BrowserObjectTypeInfo = { title: 'File', icon: fileIcon };
