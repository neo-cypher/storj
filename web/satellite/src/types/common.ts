// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export enum SortDirection {
    ASCENDING = 1,
    DESCENDING,
    asc = 1,
    desc = 2,
}

export enum OnboardingOS {
    WINDOWS = 'windows',
    MAC = 'macos',
    LINUX = 'linux',
}

export enum OnboardingOption {
    Browser = 'Continue in Browser',
    CLI = 'CLI',
    Skip = 'Skip',
}

export class PricingPlanInfo {
    constructor(
        public type: PricingPlanType = PricingPlanType.FREE,
        // Info for the pricing plan container
        public title: string = '',
        public containerSubtitle: string = '',
        public containerDescription: string = '',
        public containerFooterHTML: string | null = null,
        public activationButtonText: string | null = null,
        // Info for the pricing plan modal (pre-activation)
        public activationSubtitle: string | null = null,
        public activationDescriptionHTML: string = '',
        public activationPriceHTML: string | null = null,
        // Info for the pricing plan modal (post-activation)
        public successSubtitle: string = '',
    ) {}
}

export enum PricingPlanType {
    FREE = 'free',
    PARTNER = 'partner',
    PRO = 'pro',
}

// TODO: fully implement these types and their methods according to their Go counterparts
export type UUID = string
export type MemorySize = string
export type Time = string

export function tableSizeOptions(itemCount: number, isObjectBrowser = false): {title: string, value: number}[] {
    const opts = [
        { title: '10', value: 10 },
        { title: '25', value: 25 },
        { title: '50', value: 50 },
        { title: '100', value: 100 },
    ];
    if (itemCount < 1000 && !isObjectBrowser) {
        return [{ title: 'All', value: itemCount }, ...opts];
    }
    return opts;
}
