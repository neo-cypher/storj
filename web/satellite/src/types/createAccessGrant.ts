// Copyright (C) 2023 Storj Labs, Inc.
// See LICENSE for copying information.

import CreateNewAccessIcon from '@/../static/images/accessGrants/newCreateFlow/createNewAccess.svg';
import ChoosePermissionIcon from '@/../static/images/accessGrants/newCreateFlow/choosePermission.svg';
import AccessEncryptionIcon from '@/../static/images/accessGrants/newCreateFlow/accessEncryption.svg';
import PassphraseGeneratedIcon from '@/../static/images/accessGrants/newCreateFlow/passphraseGenerated.svg';
import AccessCreatedIcon from '@/../static/images/accessGrants/newCreateFlow/accessCreated.svg';
import CLIAccessCreatedIcon from '@/../static/images/accessGrants/newCreateFlow/cliAccessCreated.svg';
import CredentialsCreatedIcon from '@/../static/images/accessGrants/newCreateFlow/credentialsCreated.svg';
import EncryptionInfoIcon from '@/../static/images/accessGrants/newCreateFlow/encryptionInfo.svg';
import ConfirmDetailsIcon from '@/../static/images/accessGrants/newCreateFlow/confirmDetails.svg';
import TypeIcon from '@/../static/images/accessGrants/newCreateFlow/typeIcon.svg';
import NameIcon from '@/../static/images/accessGrants/newCreateFlow/nameIcon.svg';
import PermissionsIcon from '@/../static/images/accessGrants/newCreateFlow/permissionsIcon.svg';
import BucketsIcon from '@/../static/images/accessGrants/newCreateFlow/bucketsIcon.svg';
import EndDateIcon from '@/../static/images/accessGrants/newCreateFlow/endDateIcon.svg';
import EncryptionPassphraseIcon from '@/../static/images/accessGrants/newCreateFlow/encryptionPassphraseIcon.svg';

export interface IconAndTitle {
    icon: string;
    title: string;
}

export enum AccessType {
    APIKey = 'API Key',
    S3 = 'S3 Credentials',
    AccessGrant = 'Access Grant',
}

export enum PassphraseOption {
    UseExistingPassphrase = 'useExistingPassphrase',
    SetMyProjectPassphrase = 'setMyProjectPassphrase',
    GenerateNewPassphrase = 'generateNewPassphrase',
    EnterNewPassphrase = 'enterNewPassphrase',
}

export enum CreateAccessStep {
    CreateNewAccess = 'createNewAccess',
    ChoosePermission = 'choosePermission',
    EncryptionInfo = 'encryptionInfo',
    AccessEncryption = 'accessEncryption',
    PassphraseGenerated = 'passphraseGenerated',
    EnterMyPassphrase = 'enterMyPassphrase',
    EnterNewPassphrase = 'enterNewPassphrase',
    ConfirmDetails = 'confirmDetails',
    AccessCreated = 'accessCreated',
    CLIAccessCreated = 'cliAccessCreated',
    CredentialsCreated = 'credentialsCreated',
}

export enum Permission {
    All = 'all',
    Read = 'Read',
    Write = 'Write',
    List = 'List',
    Delete = 'Delete',
}

export const STEP_ICON_AND_TITLE: Record<CreateAccessStep, IconAndTitle> = {
    [CreateAccessStep.CreateNewAccess]: {
        icon: CreateNewAccessIcon,
        title: 'Create a new access',
    },
    [CreateAccessStep.ChoosePermission]: {
        icon: ChoosePermissionIcon,
        title: 'Choose permissions',
    },
    [CreateAccessStep.EncryptionInfo]: {
        icon: EncryptionInfoIcon,
        title: 'Encryption information',
    },
    [CreateAccessStep.AccessEncryption]: {
        icon: AccessEncryptionIcon,
        title: 'Access encryption',
    },
    [CreateAccessStep.PassphraseGenerated]: {
        icon: PassphraseGeneratedIcon,
        title: 'Passphrase generated',
    },
    [CreateAccessStep.EnterMyPassphrase]: {
        icon: AccessEncryptionIcon,
        title: 'Enter my passphrase',
    },
    [CreateAccessStep.EnterNewPassphrase]: {
        icon: AccessEncryptionIcon,
        title: 'Enter a new passphrase',
    },
    [CreateAccessStep.ConfirmDetails]: {
        icon: ConfirmDetailsIcon,
        title: 'Confirm details',
    },
    [CreateAccessStep.AccessCreated]: {
        icon: AccessCreatedIcon,
        title: 'Access created',
    },
    [CreateAccessStep.CredentialsCreated]: {
        icon: CredentialsCreatedIcon,
        title: 'Credentials created',
    },
    [CreateAccessStep.CLIAccessCreated]: {
        icon: CLIAccessCreatedIcon,
        title: 'CLI access created',
    },
};

export enum FunctionalContainer {
    Type = 'type',
    Name = 'name',
    Permissions = 'permissions',
    Buckets = 'buckets',
    EndDate = 'endDate',
    EncryptionPassphrase = 'encryptionPassphrase',
}

export const FUNCTIONAL_CONTAINER_ICON_AND_TITLE: Record<FunctionalContainer, IconAndTitle> = {
    [FunctionalContainer.Type]: {
        icon: TypeIcon,
        title: 'Type',
    },
    [FunctionalContainer.Name]: {
        icon: NameIcon,
        title: 'Name',
    },
    [FunctionalContainer.Permissions]: {
        icon: PermissionsIcon,
        title: 'Permissions',
    },
    [FunctionalContainer.Buckets]: {
        icon: BucketsIcon,
        title: 'Buckets',
    },
    [FunctionalContainer.EndDate]: {
        icon: EndDateIcon,
        title: 'End Date',
    },
    [FunctionalContainer.EncryptionPassphrase]: {
        icon: EncryptionPassphraseIcon,
        title: 'Encryption Passphrase',
    },
};
