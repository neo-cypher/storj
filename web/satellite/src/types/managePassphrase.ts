// Copyright (C) 2023 Storj Labs, Inc.
// See LICENSE for copying information.

export enum ManageProjectPassphraseStep {
    ManageOptions = 'ManageOptions',
    Create = 'Create',
    Switch = 'Switch',
    Clear = 'Clear',
}
