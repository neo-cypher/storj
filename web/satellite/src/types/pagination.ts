// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export const DEFAULT_PAGE_LIMIT = 10;

export type PageChangeCallback = (page: number, limit: number) => Promise<void> | void;

export type OnPageClickCallback = (search: number) => Promise<void>;
export type CheckSelected = (index: number) => boolean;

/**
 * Describes paginator page.
 */
export class Page {
    private readonly pageIndex: number = 1;
    private readonly onClick: OnPageClickCallback;

    constructor(index: number, callback: OnPageClickCallback) {
        this.pageIndex = index;
        this.onClick = callback;
    }

    public get index(): number {
        return this.pageIndex;
    }

    public async select(): Promise<void> {
        await this.onClick(this.pageIndex);
    }
}
