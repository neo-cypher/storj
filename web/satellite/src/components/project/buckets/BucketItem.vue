// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="container">
        <div class="container__header" :title="itemData.name">
            <BucketIcon class="container__header__icon" />
            <p class="container__header__name">{{ itemData.name }}</p>
        </div>
        <div class="container__item">
            <div class="container__item__inner">
                <p class="container__item__inner__label">Storage</p>
                <p class="container__item__inner__value">{{ itemData.storage.toFixed(2) }}GB</p>
            </div>
            <div class="container__item__inner">
                <p class="container__item__inner__label">Bandwidth</p>
                <p class="container__item__inner__value">{{ itemData.egress.toFixed(2) }}GB</p>
            </div>
        </div>
        <div class="container__item">
            <div class="container__item__inner">
                <p class="container__item__inner__label">Objects</p>
                <p class="container__item__inner__value">{{ itemData.objectCount.toString() }}</p>
            </div>
            <div class="container__item__inner">
                <p class="container__item__inner__label">Segments</p>
                <p class="container__item__inner__value">{{ itemData.segmentCount.toString() }}</p>
            </div>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import { Bucket } from '@/types/buckets';

import BucketIcon from '@/../static/images/project/bucket.svg';

// @vue/component
@Component({
    components: {
        BucketIcon,
    },
})
export default class BucketItem extends Vue {
    @Prop({ default: () => new Bucket('', 0, 0, 0, 0, new Date(), new Date()) })
    private readonly itemData: Bucket;
}
</script>

<style scoped lang="scss">
    .container {
        padding: 20px 40px;
        outline: none;
        display: flex;
        margin-bottom: 1px;
        box-sizing: border-box;
        font-family: 'font_medium', sans-serif;
        font-size: 16px;

        &__header {
            width: 100%;
            display: flex;
            align-items: center;

            &__icon {
                display: none;
                margin-right: 10px;
            }

            &__name {
                white-space: nowrap;
                overflow: hidden;
                text-overflow: ellipsis;
            }
        }

        &__item {
            display: flex;
            width: 100%;

            &__inner {
                width: 50%;

                &__label {
                    display: none;
                    margin-bottom: 4px;
                    font-family: 'font_regular', sans-serif;
                    font-size: 14px;
                }
            }
        }
    }

    @media screen and (max-width: 960px) {

        .container {
            flex-wrap: wrap;
            padding: 20px 24px;

            &__header {

                &__icon {
                    display: block;
                }

                &__name {
                    font-family: 'font_bold', sans-serif;
                }
            }

            &__item {
                width: 50%;
                margin-top: 16px;

                &__inner__label {
                    display: block;
                }
            }
        }
    }

    @media screen and (max-width: 600px) {

        .container {
            flex-direction: column;

            &__item {
                width: 100%;
            }
        }
    }
</style>
