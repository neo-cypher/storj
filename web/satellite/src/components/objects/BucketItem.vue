// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <table-item
        :item="itemToRender"
        :on-click="onClick"
    >
        <th slot="options" v-click-outside="closeDropdown" class="bucket-item__functional options overflow-visible" @click.stop="openDropdown(dropdownKey)">
            <dots-icon />
            <div v-if="isDropdownOpen" class="bucket-item__functional__dropdown">
                <div class="bucket-item__functional__dropdown__item" @click.stop="onDetailsClick">
                    <details-icon />
                    <p class="bucket-item__functional__dropdown__item__label">View Bucket Details</p>
                </div>
                <div class="bucket-item__functional__dropdown__item" @click.stop="onDeleteClick">
                    <delete-icon />
                    <p class="bucket-item__functional__dropdown__item__label">Delete Bucket</p>
                </div>
            </div>
            <div v-if="shouldShowGuide" class="bucket-item__functional__message" @click.stop>
                <p class="bucket-item__functional__message__title">Upload</p>
                <p class="bucket-item__functional__message__content">To upload files, open an existing bucket or create a new one.</p>
                <div class="bucket-item__functional__message__actions">
                    <div class="bucket-item__functional__message__actions__primary" @click.stop="hideGuidePermanently">
                        <p class="bucket-item__functional__message__actions__primary__label">OK</p>
                    </div>
                </div>
            </div>
        </th>
    </table-item>
</template>

<script lang="ts">
import { Component, Prop } from 'vue-property-decorator';

import { RouteConfig } from '@/router';
import { Bucket } from '@/types/buckets';
import { LocalData } from '@/utils/localData';

import TableItem from '@/components/common/TableItem.vue';
import Resizable from '@/components/common/Resizable.vue';

import DeleteIcon from '@/../static/images/objects/delete.svg';
import DetailsIcon from '@/../static/images/objects/details.svg';
import DotsIcon from '@/../static/images/objects/dots.svg';

// @vue/component
@Component({
    components: {
        TableItem,
        DotsIcon,
        DeleteIcon,
        DetailsIcon,
    },
})
export default class BucketItem extends Resizable {
    @Prop({ default: null })
    public readonly itemData: Bucket;
    @Prop({ default: () => () => {} })
    public readonly showDeleteBucketPopup: () => void;
    @Prop({ default: () => () => {} })
    public readonly openDropdown;
    @Prop({ default: () => (_: string) => {} })
    public readonly onClick: (bucket: string) => void;
    @Prop({ default: false })
    public readonly isDropdownOpen: boolean;
    @Prop({ default: true })
    public readonly showGuide: boolean;
    @Prop({ default: -1 })
    public readonly dropdownKey: number;

    public isGuideShown = true;

    public mounted(): void {
        this.isGuideShown = !LocalData.getBucketGuideHidden();
    }

    public get shouldShowGuide(): boolean {
        return this.showGuide && this.isGuideShown;
    }

    /**
     * Returns formatted date.
     */
    public get formattedDate(): string {
        return this.itemData.since.toLocaleString() || '';
    }

    public get itemToRender(): { [key: string]: string | string[] } {
        if (!this.isMobile) return { name: this.itemData.name, date: this.formattedDate };

        return { info: [ this.itemData.name, `Created ${this.formattedDate}` ] };
    }

    /*
    * Permanently hide the upload guide
    * */
    public hideGuidePermanently(): void {
        this.isGuideShown = false;
        LocalData.setBucketGuideHidden();
    }

    /**
     * Closes dropdown.
     */
    public closeDropdown(): void {
        this.openDropdown(-1);
    }

    /**
     * Holds on delete click logic.
     */
    public onDeleteClick(): void {
        this.showDeleteBucketPopup();
        this.closeDropdown();
    }

    /**
     * Redirects to bucket details page.
     */
    public onDetailsClick(): void {
        this.$router.push({
            name: RouteConfig.Buckets.with(RouteConfig.BucketsDetails).name,
            params: {
                bucketName: this.itemData.name,
                backRoute: this.$route.name || '',
            },
        });

        this.closeDropdown();
    }
}
</script>

<style scoped lang="scss">
    .bucket-item {

        &__functional {
            padding: 0 10px;
            position: relative;
            cursor: pointer;

            &__dropdown {
                position: absolute;
                top: 25px;
                right: 15px;
                background: #fff;
                box-shadow: 0 20px 34px rgb(10 27 44 / 28%);
                border-radius: 6px;
                width: 255px;
                padding: 10px 0;
                z-index: 100;

                &__item {
                    display: flex;
                    align-items: center;
                    padding: 20px 25px;
                    width: calc(100% - 50px);

                    &__label {
                        margin: 0 0 0 10px;
                    }

                    &:hover {
                        background-color: #f4f5f7;
                        font-family: 'font_medium', sans-serif;

                        svg :deep(path) {
                            fill: #0068dc;
                            stroke: #0068dc;
                        }
                    }
                }
            }

            &__message {
                position: absolute;
                top: 80%;
                width: 25rem;
                display: flex;
                flex-direction: column;
                align-items: start;
                transform: translateX(-100%);
                background-color: #0149ff;
                text-align: center;
                border-radius: 8px;
                box-sizing: border-box;
                padding: 20px;
                z-index: 1001;

                @media screen and (max-width: 320px) {
                    transform: translateX(-80%);
                }

                @media screen and (max-width: 375px) and (min-width: 350px) {
                    transform: translateX(-88%);
                }

                &:after {
                    content: '';
                    position: absolute;
                    bottom: 100%;
                    left: 10%;
                    border-width: 5px;
                    border-style: solid;
                    border-color: #0149ff transparent transparent;
                    transform: rotate(180deg);

                    @media screen and (max-width: 550px) {
                        left: 45%;
                    }
                }

                &__title {
                    font-weight: 400;
                    font-size: 12px;
                    line-height: 18px;
                    color: white;
                    opacity: 0.5;
                    margin-bottom: 4px;
                }

                &__content {
                    color: white;
                    font-weight: 500;
                    font-size: 15px;
                    line-height: 24px;
                    margin-bottom: 8px;
                    white-space: initial;
                    text-align: start;
                }

                &__actions {
                    display: flex;
                    justify-content: end;
                    align-items: center;
                    width: 100%;

                    &__primary {
                        padding: 6px 12px;
                        border-radius: 8px;
                        background-color: white;
                        margin-left: 10px;

                        &__label {
                            font-weight: 700;
                            font-size: 13px;
                            line-height: 20px;
                            color: #0149ff;
                        }
                    }
                }
            }
        }

        &:hover {
            background-color: #e6e9ef;
        }
    }
</style>
