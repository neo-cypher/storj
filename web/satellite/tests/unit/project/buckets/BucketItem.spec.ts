// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { createLocalVue, shallowMount } from '@vue/test-utils';

import { Bucket } from '@/types/buckets';

import BucketItem from '@/components/project/buckets/BucketItem.vue';

const localVue = createLocalVue();

const bucket = new Bucket('name', 1, 1, 1, 1, new Date(), new Date());

describe('BucketItem.vue', () => {
    it('renders correctly', (): void => {
        const wrapper = shallowMount(BucketItem, {
            localVue,
            propsData: {
                itemData: bucket,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
