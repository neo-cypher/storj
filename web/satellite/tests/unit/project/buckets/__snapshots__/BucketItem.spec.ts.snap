// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BucketItem.vue renders correctly 1`] = `
<div class="container">
  <div title="name" class="container__header">
    <bucketicon-stub class="container__header__icon"></bucketicon-stub>
    <p class="container__header__name">name</p>
  </div>
  <div class="container__item">
    <div class="container__item__inner">
      <p class="container__item__inner__label">Storage</p>
      <p class="container__item__inner__value">1.00GB</p>
    </div>
    <div class="container__item__inner">
      <p class="container__item__inner__label">Bandwidth</p>
      <p class="container__item__inner__value">1.00GB</p>
    </div>
  </div>
  <div class="container__item">
    <div class="container__item__inner">
      <p class="container__item__inner__label">Objects</p>
      <p class="container__item__inner__value">1</p>
    </div>
    <div class="container__item__inner">
      <p class="container__item__inner__label">Segments</p>
      <p class="container__item__inner__value">1</p>
    </div>
  </div>
</div>
`;
